Sources/App/CRP/SlackCommand+CRP.swift

```diff
@@ -69,3 +68,41 @@ extension SlackCommand {
         })
     }
 }
+
+/// Filter the CHANGELOG entries then orders and format the CHANGELOG text
+///
+/// - Parameters:
+///   - commits: The list of commits gathered between last release and current one
+///   - release: The release for which to build the CHANGELOG text for
+/// - Returns: The text containing the filtered and formatted CHANGELOG, grouped and ordered by jira board
+private func formatChangelog(using commits: [String], for release: GitHubService.Release) -> String {
+    // Only keep SDK commits if release is for SDK
+    let filteredMessages = release.isSDK ? commits.filter { $0.contains("#SDK") || $0.contains("[SDK-") } : commits
+
+    // Group then sort the changes by JIRA boards (unclassified last)
+    let grouped = Dictionary(grouping: filteredMessages) { ticket(from: $0)?.board }
+        .sorted { e1, e2 in e1.key ?? "ZZZ" < e2.key ?? "ZZZ" }
+
+    // Build a CHANGELOG text
+    return grouped
+        .reduce(into: [], { (accum: inout [String], entry: (key: String?, value: [String])) in
+            let title = entry.key.map { "\($0) tickets" } ?? "Other"
+            accum.append("## \(title)")
+            accum.append("")
```

Comment:
I'm not sure I understand though why we are appending empty strings?

Reply:
It's to end up with empty lines in the generated "markdown" (note that depending on markdown flavors, some markdown renderers require that any section title has to be separated from the section content / first paragraph by a blank line. That's not enforced by GFM which is lenient in that regard, but required by others – and tbh I personally find it nicer to read too)

Note though that I haven't tested how that will render in JIRA. Maybe I'll need to use things like `h1. ` instead of `##` for section titles? Or build a `JIRAFields.TextArea` content from scratch if I want it to be nicely formatted in JIRA? But I think a markdown-like formatting for our CRP ticket, even if section titles are not rendered by JIRA, is still readable as long as we have those newlines for readability

_PS: If you know a way in the JIRA API to send markdown text and it gets interpreted by the API to be rich text – like when you enter it from the web interface and it reformats it – instead of us having to build the `JIRAFields.TextArea` content ourselves to send to the API, I'm interested. But until then I'm hoping markdown-like is readable enough._

Reply:
won't doing `joined("\n\n")` result in the same without these empty strings?

Reply:
No, because `joined("\n\n")` would happend an empty line between _each_ line. Here I only want a blank line between the section title and the section content. Note that the `accum` is just a `[String]` (not a `[[String]]` that we'd `flatMap` later for example), and that we add all the commit entries for a given board using `append(contentOf:)`. So here we add one line (section title), then one blank line (`""`), then lots of lines at once for all the commits of that board/section (`append(contentOf:)`)